sdk/communication/communication-short-codes/test/public/manageUSProgramBriefs.spec.ts

```diff
@@ -27,80 +31,155 @@ describe(`ShortCodesClient - creates, gets, updates, lists, and deletes US Progr
+    assertEditableFieldsAreEqual(uspb, getRes, "get after update");
+  };
+
+  const _listUSProgramBriefs = async (
+    expectedProgramBriefs: USProgramBrief[],
+    byPage?: boolean,
+    itemsPerPage?: number
+  ): Promise<number> => {
+    let totalNumberOfbriefs = 0;
+    // create map of expected ids
+    const expectedBriefMap: Record<string, { brief: USProgramBrief; found: boolean }> = {};
+    expectedProgramBriefs.forEach((pb) => {
+      expectedBriefMap[pb.id] = { brief: pb, found: false };
+    });
+    if (byPage) {
+      const pages = client.listUSProgramBriefs({ top: itemsPerPage }).byPage();
+      for await (const page of pages) {
+        // loop over each item in the page
+        for (const pb of page) {
+          totalNumberOfbriefs++;
+          if (expectedBriefMap[pb.id]) {
+            expectedBriefMap[pb.id].found = true;
+            assertEditableFieldsAreEqual(
+              pb,
+              expectedBriefMap[pb.id].brief,
+              "list all program briefs, byPage"
+            );
+          }
+        }
+      }
+    } else {
       // list program briefs, validate test program brief is in the list
-      let foundTestProgramBrief = false;
       for await (const pb of client.listUSProgramBriefs()) {
-        if (pb.id === uspb.id) {
-          foundTestProgramBrief = true;
-          assertEditableFieldsAreEqual(uspb, pb, "list all program briefs");
+        totalNumberOfbriefs++;
+        if (expectedBriefMap[pb.id]) {
+          expectedBriefMap[pb.id].found = true;
+          assertEditableFieldsAreEqual(
+            pb,
+            expectedBriefMap[pb.id].brief,
+            "list all program briefs, byPage"
+          );
         }
       }
-      assert.isTrue(
-        foundTestProgramBrief,
-        "Test program brief was not returned in list of all program briefs"
-      );
+    }
+    // make sure all expected briefs were found
+    const allExpectedBriefsFound = Object.values(expectedBriefMap).every((expectedPB) => {
+      return expectedPB.found;
+    });
+    assert.isTrue(
+      allExpectedBriefsFound,
+      "Test program brief was not returned in list of all program briefs"
+    );
+    return totalNumberOfbriefs;
+  };
 
-      // delete program brief, ensure it was removed
-      const delRes = await client.deleteUSProgramBrief(uspb.id);
-      assert.isOk(delRes, "Deleting program brief failed");
-      assert.isFalse(
-        await doesProgramBriefExist(client, uspb.id),
-        "Delete program brief was unsuccessful, program brief is still returned"
-      );
+  const _testListUSProgramBriefs = async (
+    expectedProgramBriefs: USProgramBrief[]
+  ): Promise<boolean> => {
+    // list program briefs, validate test program brief is in the list
+    const totalNumberOfbriefs = await _listUSProgramBriefs(expectedProgramBriefs);
+
+    // test pagination, using 2 pages
+    const itemsPerPage = Math.floor(totalNumberOfbriefs / 2);
```

Comment:
IF totalNumberOfBriefs = 1, isn't this 0?

Reply:
fixed